.github/workflows/changelog-label.yml

```diff
@@ -0,0 +1,14 @@
+name: PR changelog label
+
+on:
+  pull_request:
+    types: [labeled, unlabeled, opened, edited, synchronize]
```

Comment:
This should use `pull_request_target` as the event type so that it runs even when the branch has conflicts. Also I think running on `edited` and `synchronize` isn't strictly necessary for a label check.